src/amo/components/Collection/index.js

```diff
@@ -79,6 +79,13 @@ export class CollectionBase extends React.Component {
       }
     }
 
+    if (collection && (
+      collection.slug !== params.slug ||
+      collection.authorUsername !== params.user
+    )) {
+      collectionChanged = true;
+    }
```

Comment:
Ugh, the loading logic in this component is really complicated. It's my first time seeing it. I know it's necessary because of SSR though. Your new logic looks fine, I'm just complaining out loud :/